parser.py

```diff
@@ -15,14 +31,11 @@ def parse_main_page(page: str) -> Generator[dict, None, None]:
     jobs_items = soup.find_all('article', class_='FxQpvm yKsady')
 
     for job_item in jobs_items:
-        job_name = job_item.find('span', class_='_3862j6').find('span').text.strip()
-        job_url = job_item.find('div', class_='_15V35X').find('a').get('href')
-        salary_tag = job_item.find('p', class_='jNebTl')
-        salary = None
-        if salary_tag is not None:
-            salary = salary_tag.text.strip()
-        short_description = job_item.find('div', class_='_9jGwm1').text.strip()
-        yield create_item_main_page(job_name, job_url, salary, short_description)
+        job_name = soup_find_exception_checker(job_item.find('span', class_=soup_class_dict['job_name']).find('span'))
+        job_url = soup_find_exception_checker(job_item.find('div', class_=soup_class_dict['job_url']).find('a').get('href'))
```

Comment:
It seems that we still can have here exception, if first `find` in chain will return `None`

Reply:
done